src/Service/CertificateLocator.php

```diff
@@ -66,4 +69,20 @@ public function getCertificateLocator(): CertificateLocatorInterface
 
         throw new CertificateLocatorException(sprintf('Invalid certificate locator type: %s', $locatorType));
     }
+
+    /**
+     * Checks if certificate locator token should be refreshed.
+     */
+    public function shouldAzureKeyVaultAccessTokenBeRefreshed(): bool
```

Comment:
Consider renaming this function to `isExpired` or `needsRefresh`, i.e. don't mention “Azure Key Vault”. Other (theoretical/future) locators may need refreshment as well.

Reply:
Renamed to `needsRefresh`.